app_dart/lib/src/service/scheduler.dart

```diff
@@ -466,6 +469,33 @@ class Scheduler {
     return true;
   }
 
+  /// Given a [checkRunEvent], finds the [PullRequest] that the check run must belong to.
+  Future<PullRequest?> _findMatchingPullRequest(cocoon_checks.CheckRunEvent checkRunEvent) async {
```

Comment:
Does this belong to the github API? I see potential use cases for it in the future and it won't be intuitive to use from here.

Reply:
That sounds good. I'll shift it over.